api/src/pcapi/routes/backoffice/autocomplete.py

```diff
@@ -209,6 +220,35 @@ def autocomplete_pricing_points() -> AutocompleteResponse:
     return _autocomplete_venues(only_with_siret=True)
 
 
+@blueprint.backoffice_web.route("/autocomplete/providers", methods=["GET"])
+@spectree_serialize(response_model=AutocompleteResponse, api=blueprint.backoffice_web_schema)
+def autocomplete_providers() -> AutocompleteResponse:
+    query_string = request.args.get("q", "").strip()
+    is_numeric_query = query_string.isnumeric()
+
+    query = providers_models.Provider.query.options(
+        sa.orm.load_only(providers_models.Provider.id, providers_models.Provider.name)
+    )
+
+    if not is_numeric_query and len(query_string) < 2:
+        return AutocompleteResponse(items=[])
+
+    if is_numeric_query and len(query_string) == 1:
+        filters = providers_models.Provider.id == int(query_string)
+    else:
+        or_filters = [
+            sa.func.unaccent(providers_models.Provider.name).ilike(f"%{clean_accents(query_string)}%"),
+        ]
+        if is_numeric_query and len(query_string) <= 3:
+            or_filters.append(providers_models.Provider.id == int(query_string))
```

Comment:
Je ne saisis pas trop, je pense qu'on peut simplifier car en pratique, aucun nom ne contient de chiffre.
L'ensemble pourrait donc être : 
```
    if is_numeric_query:
        filter = providers_models.Provider.id == int(query_string)
    else:
        filter = sa.func.unaccent(providers_models.Provider.name).ilike(f"%{clean_accents(query_string)}%")
```
jusqu'à preuve d'un besoin plus sophistiqué :)

Reply:
J'approuve... et pas seulement parce que je ne comprends pas bien la logique basée sur le nombre de caractères. :)

Reply:
Je pense cette logique est un reliquat repris d'un copier/coller de l'autocomplétion de structures et de lieux, avec des conditions respectivement `if is_numeric_query and len(query_string) <= 9:` et `if is_numeric_query and len(query_string) <= 14:` pour chercher aussi dans les SIREN et SIRET. On n'a a priori pas d'équivalent ici.

Reply:
j'ai simpléfié et intégré la gestion d'une requete trop courte dans le même if.

Reply:
Finalement, tu as conservé la condition `and len(query_string) <= 3:` ? J'étais d'accord avec la suggestion de Patrick de la supprimer, elle n'a pas l'air d'apporter grand chose. Qu'en penses-tu ?

Reply:
... hmm, en fait, ici, on est dans un bloc où `is_numeric_query` est faux (sinon on serait passé dans le bloc `if` en ligne 237). On peut donc supprimer tout ce bloc : 

```python
if is_numeric_query and len(query_string) <= 3:
    or_filters.append(providers_models.Provider.id == int(query_string))
```

... non ?

Reply:
oui, je ne ais pas comment il a survecu dsl. Je le surpimme (et je refactor un peu le code pour le simplifier)